### PR TITLE
feat: implemented SpawnerConfiguration

### LINKED ISSUE #111: Add `SpawnerConfiguration` class
## User stories
As a user, that wants to configure a `Spawner` I want to know the configuration fields the  in order to add it.

## (Wiki) References
- https://github.com/BP2022-AP1/bp2022-ap1/wiki/Architecture#database-class-diagram

## Acceptance Criteria
- [ ] Add python class with peewee and marshmellow, without migrations and foreign keys
- [ ] Update database class diagram in wiki

### PR DESCRIPTION
**Pull request by CR1337**
Fixes #111

Implemented `SpawnerConfiguration`. 

A `SpawnerConfiguration` can have many `Schedule`s and a `Schedule` ca be part of many `SpawnerConfiguration`s. Therefore I implemented a m:n relation with `SpawnerConfigurationXSchedule`

## PR checklist

- [x] Acceptance criteria fulfilled
- [x] Additional features are tested
- [x] Docs (code, wiki & diagrams) updated
- [ ] Breaking changes anounced
- [ ] Dev-branch has been merged into local branch to resolve conflicts
- [ ] Tests and linter have passed AFTER local merge
- [ ] Another dev reviewed and approved

### CODE REVIEW

**Comment by CR1337** [comment]
Please ignore `class Spawner`. I will deal with it in a later issue.

File: src/spawner/spawner.py [hunk 1]
```diff
@@ -1,9 +1,56 @@
...
+from peewee import ForeignKeyField
+
+from src.base_model import BaseModel
 from src.component import Component
 from src.schedule.schedule import Schedule
 
 
+class SpawnerConfiguration(BaseModel):
+    """Class representing a spawner configuration. It holds a list of
+    Schedules which is handled in the reference table class `SpawnerConfigurationXSchedule`.
```
Comment (SaturnHafen) [inline src/spawner/spawner.py]:
```suggestion
    Schedules which are handled in the reference table class `SpawnerConfigurationXSchedule`.
```

File: src/spawner/spawner.py [hunk 2]
```diff
@@ -1,9 +1,56 @@
...
+
+            :param data: The dictionary.
+            :return: A SpawnerConfigurationXSchedule.
+            """
+            return SpawnerConfigurationXSchedule(**data)
+
+    spawner_configuration_id = ForeignKeyField(
+        SpawnerConfiguration, null=False, backref="schedules"
+    )
+    schedule_id = ForeignKeyField(Schedule, null=False)
```
Comment (SaturnHafen) [inline src/spawner/spawner.py:51]: Does this mean that a spawner must have at least one schedule (and the reverse is also true)? If yes, does this work with the API (can you create a spawner config and a schedule simultaneously?)

**Review by SaturnHafen — COMMENTED** [review]
Looks good to me. I don't know enough about the database stuff to approve though.

File: src/spawner/spawner.py (hunk 2 above)
Comment (antonykamp) [reply src/spawner/spawner.py:51]: Interesting Question! 👀

**Review by antonykamp — COMMENTED** [review]
Don't forget to test your beautiful code ;-)

File: src/spawner/spawner.py (hunk 2 above)
Comment (CR1337) [reply src/spawner/spawner.py:51]: A `SpawnerConfiguration` can have 0 or more schedules. For each `Schedule` it has, there exists a `SpawnerConfigurationXSchedule` that holds the corresponding `schedule_id` and `spawner_configuration_id`. It is just a m:n relation.

**Comment by CR1337** [comment]
> Don't forget to test your beautiful code ;-)

Is there a short tutorial for our beautiful database- and marshmallow test code? :)

**Comment by antonykamp** [comment]
> Is there a short tutorial for our beautiful database- and marshmallow test code? :)

Just add failing your class to the parametrization ;) Add invalid dicts to the `TestFailingDict` class and valid dicts to `TestCorrectFilledDict` class 😇 We recommend using dict with all fields filled and a dict with only required fields filled in the last class 😄 

Do you think I should add more documentation that the class docstring? 🤔

**Comment by CR1337** [comment]
Testing makes problems. Will be tackled in another issue.

**Review by antonykamp — APPROVED** [review]
Will be fixed in later issue/pr

**Pull request merged**